feeds/wlan-ap/opensync/src/platform/openwrt/src/netifd/src/wifi_inet_state.c

```diff
@@ -65,19 +65,36 @@ static const struct blobmsg_policy route_policy[__ROUTE_ATTR_MAX] = {
 static ovsdb_table_t table_Wifi_Inet_State;
 static ovsdb_table_t table_Wifi_Master_State;
 
-int l3_device_split(char *l3_device, struct iface_info *info)
+int l3_device_split_gre(char *l3_device, struct iface_info *info)
 {
-	char *dot;
+	char *delim;
+	memset (info, 0, sizeof(*info));
+	delim = strstr(l3_device, ".");
+	if (!delim) {
+		strcpy(info->name, &l3_device[6]);
+	} else {
+		strncpy(info->name, &l3_device[6], delim - l3_device - 6);
+		info->vid = atoi(&delim[1]);
+	}
+
+	return 0;
+}
 
+int l3_device_split(char *l3_device, struct iface_info *info)
+{
+	char *delim;
 	if (strncmp(l3_device, "br-", 3))
 		return -1;
 	memset (info, 0, sizeof(*info));
-	dot = strstr(l3_device, ".");
-	if (!dot) {
+	if (!strncmp(l3_device, "br-gre", 6))
+		delim = strstr(l3_device, "_");
+	else
+		delim = strstr(l3_device, ".");
+	if (!delim) {
```

Comment:
Delim is "_" when "br-gre".  Is that what you want?  Seems opposite to l3_device_split_gre() added above.

Reply:
"_" is required here as we are keeping GRE bridge name(s) as br-gre_<vlan_id>  while l3_device_split_gre() handles the vlan ifname inside the GRE bridge which would be for eg: gre4t-gre.<vlan_id>

Reply:
It seems underscore is not allowed in comments. Anyhow, GRE bridge names use br-gre<underscore><vlan_id>. Therefore, the delim is used.

Reply:
Thanks